shared/middleware/channels_api.rb

```diff
@@ -59,6 +59,9 @@ class ChannelsApi < Sinatra::Base
 
     if src_channel
       data = storage_app.get(src_channel)
+      if data['name'].nil? and data['level'] == '/projects/algebra_game'
```

Comment:
Is it the case that the some past data doesn't have data['name'] populated, but in the future it will always be populated?

It would be nicest if we could make a sweep through the data (which I believe is in DynamoDB?) and backfill data['name'] as a one time job, that way don't have to worry about it every being null anywhere in the code.  Otherwise, the backfill logic needs to stay in the code forever since we can never be sure that old data isn't still hanging around.

If that's too difficult, it would be good to factor out the migration logic into a helper function (e.g. migrate_old_data) rather than including it inline in the controller logic.